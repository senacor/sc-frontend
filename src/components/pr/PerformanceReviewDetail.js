import React, { useContext, useEffect, useState } from 'react';
import CircularProgress from '@material-ui/core/CircularProgress/index';
import PrState from './PrState';
import PrTabs from './PrTabs';
import PrDetailInformation from './PrDetailInformation';
import { fetchMeeting } from '../../calls/meetings';
import { fetchPrById } from '../../calls/pr';
import { ErrorContext, MeetingContext, PrContext } from '../App';

const PerformanceReviewDetail = props => {
  const { value: pr, setValue: setPr } = useContext(PrContext.context);
  const [isLoading, setIsLoading] = useState({});
  const errorContext = useContext(ErrorContext.context);
  const { value: meeting, setValue: setMeeting } = useContext(
    MeetingContext.context
  );

  useEffect(() => {
    const afterPrFetched = pr => {
      setPr(pr);
      fetchMeeting(pr, setMeeting, errorContext);
    };
    fetchPrById(
      props.match.params.id,
      afterPrFetched,
      setIsLoading,
      errorContext
    );
  }, []);

  if (isLoading || !pr || Object.entries(pr).length === 0) {
    return <CircularProgress />;
  }

  return (
    <React.Fragment>
      {pr && <PrDetailInformation pr={pr} meeting={meeting} />}
      {pr && <PrState prById={pr} />}
      {pr && <PrTabs pr={pr} meeting={meeting} />}
    </React.Fragment>
  );
};

export default PerformanceReviewDetail;
