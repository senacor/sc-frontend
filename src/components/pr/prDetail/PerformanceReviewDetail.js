import React, { Component } from 'react';
import PrState from './PrState';
import PrTabs from './PrTabs';
import { connect } from 'react-redux';
import { getPrDetail } from '../../../reducers/selector';
import * as actions from '../../../actions';
import withLoading from '../../hoc/Loading';

export class PerformanceReviewDetail extends Component {
  render() {
    let { prById } = this.props;
    return (
      <React.Fragment>
        <PrState />
        {prById ? <PrTabs /> : null}
      </React.Fragment>
    );
  }
}

PerformanceReviewDetail.propTypes = {};

export default connect(
  state => ({
    prById: getPrDetail()(state),
    isLoading: state.isLoading
  }),
  {
    fetchPrById: actions.fetchPrById,
    fetchMeeting: actions.fetchMeeting
  }
)(
  withLoading(props => {
    return props
      .fetchPrById(props.match.params.id)
      .then(pr => props.fetchMeeting(pr));
  })(PerformanceReviewDetail)
);
