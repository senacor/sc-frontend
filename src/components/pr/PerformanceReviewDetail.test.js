import React from 'react';
import { shallow } from 'enzyme';
import { PerformanceReviewDetail } from './PerformanceReviewDetail';

describe('PerformanceReviewDetail', () => {
  it('should match snapshot', () => {
    let cut = shallow(<PerformanceReviewDetail prDetail={{}} />);

    expect(cut).toMatchSnapshot();
  });
});
