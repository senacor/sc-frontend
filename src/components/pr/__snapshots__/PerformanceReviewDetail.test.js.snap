// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerformanceReviewDetail should match snapshot 1`] = `
<WithStyles(Grid)
  container={true}
  spacing={8}
>
  <WithStyles(Grid)
    item={true}
    lg={8}
    md={8}
    sm={12}
    xl={8}
    xs={12}
  >
    <Connect(WithStyles(PrSheet)) />
  </WithStyles(Grid)>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(Grid)
      item={true}
      lg={4}
      md={4}
      xl={4}
    >
      <WithStyles(Grid)
        container={true}
        direction="row"
        spacing={8}
      >
        <WithStyles(Grid)
          item={true}
          lg={12}
          md={12}
          xl={12}
        >
          <WithStyles(PrSalary)
            prDetail={Object {}}
          />
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          lg={12}
          md={12}
          xl={12}
        >
          <WithStyles(PrEmployment)
            prById={Object {}}
          />
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          lg={12}
          md={12}
          xl={12}
        >
          <WithStyles(PrState) />
        </WithStyles(Grid)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </Hidden>
</WithStyles(Grid)>
`;
